src/meta/src/barrier/mod.rs

```diff
@@ -394,13 +412,28 @@ where
                 GlobalBarrierManager::run_inner(
                     env.clone(),
                     &command_ctx,
-                    command_ctx_queue,
+                    command_ctx_queue.clone(),
                     hummock_manager,
                     wait_recovery,
                     wait_build_actor,
                     barrier_send_tx_clone,
                 )
                 .await;
+                let command_ctx_queue = command_ctx_queue.read().await;
+                let in_flight_nums = command_ctx_queue
+                    .iter()
+                    .filter(|x| matches!(x.states, InFlight))
+                    .count();
+                metrics
+                    .barrier_nums
+                    .get_metric_with_label_values(&["in_flight"])
+                    .unwrap()
+                    .set(in_flight_nums as i64);
+                metrics
+                    .barrier_nums
+                    .get_metric_with_label_values(&["all_barrier"])
```

Comment:
If the number of series is determined at compile time (e.g. we have inflight and all barriers), it's better to create two different gauges instead of using one gauge with two labels.

Reply:
okokok!!!